include/dgl/base_heterograph.h

```diff
@@ -465,6 +465,12 @@ class BaseHeteroGraph : public runtime::Object {
     return nullptr;
   }
 
+  /*! \brief Creat a LineGraph of self */
+  virtual HeteroGraphPtr LineGraph(bool backtracking) const {
+    LOG(FATAL) << "LineGraph not supported.";
+    return nullptr;
+  }
```

Comment:
Could you make it as a global function instead of a member function? Overall, I will try make every graph transformation API as functional as possible.

Reply:
Sure.